lamindb/curators/__init__.py

```diff
@@ -198,6 +198,12 @@ def save_artifact(
         # constructor signature
         pass
 
+    def _set_dataset_and_artifact(self):
```

Comment:
The function is misnamed now because it doesn't accept `dataset` as an argument (as I thought it would).

Given it operates on the `self._dataset` attribute, it should be called `_set_artifact()` or something similar.

Overall, there is less logic here than I thought, will look further down the code.